File: algorithm/src/main/java/baekjoon/p05704/Main.java
package baekjoon.p05704;

import java.util.HashSet;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            String input = sc.nextLine();
            if (input.equals("*")) {
                break;
            }

            System.out.println(solve(input));
        }
    }

    private static String solve(String input) {
        HashSet<Character> hashSet = new HashSet<>();
        for (char ch : input.toCharArray()) {
            if (ch == ' ') {
                continue;
            }

            hashSet.add(ch);
        }

        if (hashSet.size() == 26) {
            return "Y";
        } else {
            return "N";
        }
    }

}

File: algorithm/src/main/java/baekjoon/p05704/Main.kt
package baekjoon.p05704

fun main(args: Array<String>) {
    while (true) {
        val input = readLine()!!
        if (input == "*") {
            break
        }

        println(solve(input))
    }
}

private fun solve(input: String): String {
    return when (('a'..'z').all { input.contains(it) }) {
        true -> "Y"
        false -> "N"
    }
}